### PR TITLE
Remove v62

### PR DESCRIPTION
**Pull request by Zimmi48**
This old hint database dates back from the introduction of hint databases (for easy compatibility with previous developments). Until today, the coq-contribs were still (for half of them) relying on it. Removing all its uses was a matter of replacing them with the right db, such as arith, sets, bool, or datatypes (most of the time in an automatic fashion).

This PR removes all references to this in the 8.6 codebase (most notably in the stdlib were hints are now declared in only one db, and in the refman).

If this gets merged before #332, I will update it to not introduce one more reference to `v62`.

PS: this needs to be tested against coq-contribs but ci.inria.fr/coq and coq.inria.fr seem to be experimenting problems right now (both my test and Maxime's failed because the slave went off-line).

### CODE REVIEW

**Comment by Zimmi48** [comment]
Testing here: https://ci.inria.fr/coq/view/coq-contribs/job/coq-contribs/549/

**Comment by Zimmi48** [comment]
As expected, it passed the contribs successfully. Can this get merged quickly?

**Comment by Zimmi48** [comment]
Does the same reasoning apply here? Shall we mark `v62` deprecated and shall I rebase this PR on trunk? Or do we consider that the removal of an old compatibility option can be done without deprecation warning, similarly to what was planned for the < 8.4 compatibility flags?

**Comment by mattam82** [comment]
Ha, it's on the line... Is it as easy as adding a deprecation warning at
parsing/interpretation time?

**Comment by Zimmi48** [comment]
@mattam82 What do you mean? A deprecation warning is supposed to come in a version before the feature is removed. In particular, if this removal was done in trunk only, then the deprecation warning could be in the doc of 8.6. If we remove `v62` in 8.6, well basically nothing will happen because only the contribs used it but if someone wanted to use it, it would just say `Error: this db does not exist.` and it would be consistent with the doc which would not mention it anymore.

**Comment by maximedenes** [comment]
This looks to me a bit like removing a very old compat flag, which sounds ok to do even now. I hope I'm not forgetting more serious compatibility issues.

**Comment by Zimmi48** [comment]
BTW have the old compat flags been finally removed? (8.2, 8.3)
I planned to take care of it at some point and then stopped dead in my tracks when I understood that it was too late for new clean-ups in 8.6.

**Comment by maximedenes** [comment]
I did it on a local branch, but it turned out to be a nightmare, because I needed to do something with compatibility notations in the stdlib. I'll probably say more about it during the next working group.

File: tactics/hints.ml [hunk 1]
```diff
@@ -657,8 +657,7 @@ let searchtable_add (name,db) =
 let current_db_names () = Hintdbmap.domain !searchtable
 let current_db () = Hintdbmap.bindings !searchtable
 
-let current_pure_db () =
-  List.map snd (Hintdbmap.bindings (Hintdbmap.remove "v62" !searchtable))
+let current_pure_db () = List.map snd (current_db ())
```
Comment (maximedenes) [inline tactics/hints.ml]: Out of curiosity, what is a  "pure" db?

Comment (Zimmi48) [inline tactics/hints.ml]: I don't know the origin of the name but it seems that it's just the hintdb you want. I am not sure what is stored with it. There's a bit more refactoring to do around this `hints.ml` file which I intend to do shortly (I was just expecting to see how this PR would get merged first). I am planning in particular to remove `current_db_names` from `hints.mli`.

**Pull request merged**